### PR TITLE
fix(auth): Fix issue in auth configure where it fails if both Cognito services are not present

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:* https://github.com/aws-amplify/amplify-ios/issues/586

*Description of changes:*

- Auth plugin now require at least one of the Cognito Service configuration to properly configure.
- Added unit test for different scenarios of configuration.

Dependent on - https://github.com/aws-amplify/aws-sdk-ios/pull/2812

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift [hunk 1]
```diff
@@ -62,34 +57,50 @@ extension AWSCognitoAuthPlugin {
...
+                                      identityPoolConfiguration: identityPoolConfig)
+    }
+
+    func identityPoolServiceConfiguration(from authConfiguration: JSONValue) -> AmplifyAWSServiceConfiguration? {
+        let regionKeyPath = "CredentialsProvider.CognitoIdentity.Default.Region"
+        guard case .string(let regionString) = authConfiguration.value(at: regionKeyPath) else {
+            let amplifyMessage =
+            """
+                Could not read Cognito identity pool information from the configuration.
+                """
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift:86]: nit: indentation. Do we need this to be a multiline string in the first place?

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift [hunk 2]
```diff
@@ -62,34 +57,50 @@ extension AWSCognitoAuthPlugin {
...
     }
 
-    func userPoolServiceConfiguration(from authConfiguration: JSONValue) throws -> AmplifyAWSServiceConfiguration {
+    func userPoolServiceConfiguration(from authConfiguration: JSONValue) -> AmplifyAWSServiceConfiguration? {
         let regionKeyPath = "CognitoUserPool.Default.Region"
         guard case .string(let regionString) = authConfiguration.value(at: regionKeyPath) else {
-            let amplifyError = AuthError.configuration(
-                "Error configuring \(String(describing: self))",
+            let amplifyMessage =
+            """
```
Comment (palpatim) [inline AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift:99]: Does this need to be multiline? If so, fix indentation

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift (hunk 1 above)
Comment (royjit) [reply AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift]: Fixed

File: AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift (hunk 2 above)
Comment (royjit) [reply AmplifyPlugins/Auth/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift]: Fixed

**Comment by dzeitman** [comment]
This issue still persists!

**Pull request merged**